agent/src/model.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 Oxide Computer Company
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, JsonSchema, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum State {
+    Requested,
+    Created,
+    Tombstoned,
+    Destroyed,
+    Failed,
+}
+
+#[derive(Serialize, Deserialize, JsonSchema, Debug, PartialEq, Clone)]
+pub struct Region {
```

Comment:
Seems like there could be some overlap here with the region struct and the region struct that the upstairs/downstairs uses.  Perhaps in the future (not now) we can merge these two.